net/haproxy/src/opnsense/mvc/app/controllers/OPNsense/HAProxy/forms/dialogAction.xml

```diff
@@ -23,59 +23,304 @@
         <type>dropdown</type>
-        <help><![CDATA[Use this server if the condition is true.<br/><b>Not used for any other action.</b>]]></help>
+        <help><![CDATA[HAProxy will use this server if the condition evaluates to true.]]></help>
+    </field>
+    <field>
+        <label>Parameters</label>
+        <type>header</type>
+        <style>type_table table_http-request_auth</style>
+    </field>
+    <field>
+        <id>action.http-request_auth</id>
+        <label>Auth Realm</label>
+        <type>text</type>
+        <help><![CDATA[When HAProxy requests user name and password from the user, this optiomal authentication realm is returned with the response (typically the application's name).]]></help>
+    </field>
+    <field>
+        <label>Parameters</label>
+        <type>header</type>
+        <style>type_table table_http-request_redirect</style>
+    </field>
+    <field>
+        <id>action.http-request_redirect</id>
+        <label>HTTP Redirect</label>
+        <type>text</type>
+        <help><![CDATA[Use HAProxy's powerful redirect function to return a HTTP redirection. See <a href="http://cbonte.github.io/haproxy-dconv/1.7/configuration.html#redirect">HAProxy's documentation</a> for further details and examples.]]></help>
+    </field>
+    <field>
+        <label>Parameters</label>
+        <type>header</type>
+        <style>type_table table_http-request_lua</style>
+    </field>
+    <field>
+        <id>action.http-request_lua</id>
+        <label>Lua function</label>
+        <type>text</type>
+        <help><![CDATA[Run the specified Lua function. You will most likely need to include/load your Lua code first.]]></help>
+    </field>
+    <field>
+        <label>Parameters</label>
+        <type>header</type>
+        <style>type_table table_http-request_use-service</style>
+    </field>
+    <field>
+        <id>action.http-request_use-service</id>
+        <label>Lua service</label>
+        <type>text</type>
+        <help><![CDATA[Register the specified Lua service. You will most likely need to include/load your Lua code first.]]></help>
+    </field>
+    <field>
+        <label>Parameters</label>
+        <type>header</type>
+        <style>type_table table_http-request_add-header</style>
     </field>
     <field>
-        <label>Conditional parameters</label>
+        <id>action.http-request_add-header_name</id>
+        <label>HTTP Header</label>
+        <type>text</type>
+        <help><![CDATA[Append a HTTP header field with the specified name.]]></help>
+    </field>
+    <field>
+        <id>action.http-request_add-header_content</id>
+        <label>Header Content</label>
+        <type>text</type>
+        <help><![CDATA[The value that should be set for the specified HTTP header. Note that it's possible to use pre-defined variables, see <a href="http://cbonte.github.io/haproxy-dconv/1.7/configuration.html#8.2.4">HAProxy's documentation</a> for further details and examples.]]></help>
```

Comment:
As a general note, technical documentation tries to avoid contractions as much as possible: "it's" becomes "it is" and so on